### PR TITLE
feat(node): node sdk

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

New package: node

Extended from `client`, implement 4 adapters: `requester`, `generateCodeChallenge`, `generateCodeVerifier`, `generateState`.

Leave `storage` and `navigate` to users.

## Testing

UTs.

### CODE REVIEW

File: packages/node/src/include.d/node-fetch.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+declare module 'node-fetch' {
+  const foo: typeof fetch;
+  export = foo;
```
Comment (gao-sun) [inline packages/node/src/include.d/node-fetch.d.ts:3]:
```suggestion
  const nodeFetch: typeof fetch;
  export = nodeFetch;
```

File: packages/node/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "extends": "@silverhand/ts-config/tsconfig.base",
+    "compilerOptions": {
+        "outDir": "lib",
+        "types": [
+            "jest",
+            "jest-matcher-specific-error"
```
Comment (demonzoo) [inline packages/node/tsconfig.json:7]: nit: don't we need `node` here?

**Review by gao-sun — APPROVED** [review]

**Review by demonzoo — APPROVED** [review]
LGTM

File: packages/node/src/include.d/node-fetch.d.ts (hunk 1 above)
Comment (demonzoo) [reply packages/node/src/include.d/node-fetch.d.ts:3]: +1

File: packages/node/tsconfig.json (hunk 2 above)
Comment (wangsijie) [reply packages/node/tsconfig.json:7]: is OK for now

**Pull request merged**